### PR TITLE
Add new relational node parser for markdown

### PR DESCRIPTION
**Pull request by hexapode**
# Description

Add a new relational node parser for markdown working as similarly as possible to the unstructured_node_element node parser, so that we can unify/generalize  the approach in the future.

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update ?

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by logan-markewich** [comment]
`so that we can unify/generalize the approach in the future` -- yea, it feels like there should be a base class for this soon :)

File: llama_index/node_parser/relational/markdown_element.py [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+        callback_manager = callback_manager or CallbackManager([])
+
+        return cls(
+            callback_manager=callback_manager,
+        )
+
+    def get_nodes_from_node(self, node: TextNode) -> List[BaseNode]:
+        """Get nodes from node."""
+        elements = extract_markdown_elements(
+            node.get_content(), table_filters=[filter_table]
```
Comment (logan-markewich) [inline llama_index/node_parser/relational/markdown_element.py:356]: we probably want to use `node.get_content(metadata_mode=MetadataMode.NONE)` here

Comment (hexapode) [reply llama_index/node_parser/relational/markdown_element.py]: I made the change, was under the impression that it was the default value, but apparently not for all type of Nodes

Comment (logan-markewich) [reply llama_index/node_parser/relational/markdown_element.py]: Ah it is `NONE` by default lol my bad

**Comment by logan-markewich** [comment]
@hexapode I refactored this out into a base class -- just wanted to reduce duplicated code :) I think this should lend itself well to expand to other element types in the future (figures, headings, footers, etc)

**Comment by hexapode** [comment]
> @hexapode I refactored this out into a base class -- just wanted to reduce duplicated code :) I think this should lend itself well to expand to other element types in the future (figures, headings, footers, etc)
> 
> But let me know what you think!

@logan-markewich the base class work well, I like it!

**Pull request merged**